Docs v.0.16 update: FS Browser & Cluster nodes Monitor (#810)
epam/cloud-pipeline: docs/md/manual/15_Interactive_services/15.4._Expose_node_filesystem.md

# 15.4. Expose node filesystem

- [Upload files through node's file browser](#Upload-files-through-node's-file-browser)
- [Download files through node's file browser](#Download-files-through-node's-file-browser)
- [Delete files through node's file browser](#Delete-files-through-node's-file-browser)
- [Search files at node's file browser](#Search-files-at-node's-file-browser)

> **Filesystem browser** is available for the **Active runs** only.  

> To browse node filesystem go to the runs page:

Reviewer: ```suggestion
To browse node filesystem go to the **Run logs** page.
```

>     ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_01.png)

Reviewer: Suggest not to show any pictures at this place
```suggestion
```

> After a job had been initialized the `Browse` hyperlink become available to the users via the Cloud Pipeline Web GUI within a run's log form and a dashboard tiles of the running pipelines:   

Reviewer: ```suggestion
After a job had been initialized the `Browse` hyperlink become available to the users:
```
Or what did you mean about "dashboard tiles"?

    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_02.png)

> Click on the hyperlink to view node's filesystem:

Reviewer: ```suggestion
Click the hyperlink to view the node's filesystem:  
```
Don't forget to put 2 spaces at the end of a line if you want to add the line break (without an empty line) in `Markdown` documents

    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_03.png)

User has an ability to view, download, upload, delete and search files and directories.

**_Note_**: Storage has an auto-cleanup policy (to remove the temp files).

> 

Reviewer: Add small info+picture(s) about folder navigation in the FS Browser

Author: Done

## Upload files through node's file browser

To upload files to the node filesystem press the `Upload` button:   
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_04.png)

> The local File Manager window will be opened. Select files that you want to upload and press the Open button:

Reviewer: ```suggestion
The local File Manager window will be opened. Select files that you want to upload and press the Open button:  
```

    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_05.png)

> The Cloud Pipeline will display `Operations` list with all selected files and uploading statuses:

Reviewer: ```suggestion
The Cloud Pipeline will display `Operations` list with all selected files and uploading statuses:  
```

    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_06.png)

> After the uploading will be completed the system will update files statuses: 

Reviewer: ```suggestion
After the upload is completed the system will update the file statuses:  
```

    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_07.png)

> To collapse the Operations window press cross button.

Reviewer: ```suggestion
To collapse the Operations window press cross button.  
```

You can always reopen Operations list by pressing Operations list button at the bottom corner of the page:   
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_08.png)

> ## Download files through node's file browser

Reviewer: ```suggestion
## Download files/directories through node's file browser
```
And correct, please, the corresponding item of the page content

Author: Done

Download operation supports both files and directories. If a directory is requested for a download the result of such operation will be a gzipped tarball.

To download files or directories on your local machine from the node filesystem press the `Download` button:   
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_09.png)

> 

Reviewer: Add, please, a picture with a downloaded gzipped directory. Because now, it's hard to realize why you click to download the 'cloud-data' directory but on the next picture there is only one downloaded file (not *.gzip)

Author: Done

The Operations window will be open:  

>     ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_10.png)

Reviewer: ```suggestion
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_11.png)
```

Author: Done

The file will be downloaded to your local machine after it's status in Operations window became `Downloaded`:   

>     ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_11.png)

Reviewer: ```suggestion
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_10.png)
```
Maybe this replacement is better?

Author: Done

> ## Delete files through node's file browser

Reviewer: ```suggestion
## Delete files/directories through node's file browser
```
And correct, please, the corresponding item of the page content

> To delete file or directory user needs to press the `Delete` button:

Reviewer: ```suggestion
To delete file or directory user needs to press the `Delete` button:  
```

>     ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_12.png)

Reviewer: ```suggestion
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_12.png)  
```

> The confirmation window will be displayed. Press OK to delete file or directory:

Reviewer: ```suggestion
The confirmation window will be displayed. Press the OK button to delete:
```

>     ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_13.png)'

Reviewer: ```suggestion
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_13.png)
```
Please, remove from the picture the part with the text in Russian

> ## Search files at node's file browser

Reviewer: ```suggestion
## Search files/directories at node's file browser
```
And correct, please, the corresponding item of the page content

Author: Done

> To search inside current directory through files and directories click on the address panel and type desired file or directory name:

Reviewer: ```suggestion
To search inside the current directory through files and directories - click the address panel and type desired file or directory name:  
```

>     ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_14.png)

Reviewer: ```suggestion
    ![CP_v.0.16_ReleaseNotes](attachments/NodeFilesystem_14.png)  
```

# 15.4. Expose node filesystem

> - [Upload files through node's file browser](#Upload-files-through-node's-file-browser)

Reviewer: For correct work of anchor-links in `Markdown` documents - link addresses should be written in the lowercase, spaces should be replaced by '-' symbol, other auxiliary symbols are also replaced by '-' or removed
```suggestion
- [Upload files through node's file browser](#upload-files-through-nodes-file-browser)
```
Analogically correct, please, other content items.
